torchx/lib/torchx/backend.ex

```diff
@@ -0,0 +1,323 @@
+      NIF.unquote(op)(to_ref(left), to_ref(right)) |> from_ref(out)
+    end
+  end
+
+  defp maybe_cast_u8(%T{type: {t, _}} = left, %T{type: {t, _}} = right), do: {left, right}
+
+  defp maybe_cast_u8(%T{type: {:u, 8}} = left, %T{} = right),
+       do: {Nx.as_type(left, {:s, 16}), right}
+
+  defp maybe_cast_u8(%T{} = left, %T{type: {:u, 8}} = right),
+       do: {left, Nx.as_type(right, {:s, 16})}
+
+  defp maybe_cast_u8(left, right), do: {left, right}
+
+
+  for op <- [:bitwise_and, :bitwise_or, :bitwise_xor] do
+    @impl true
+    def unquote(op)(out, l, r) do
+      {left, right} = maybe_cast_u8(l, r)
+
+      %T{type: {_, size_left}} = left
+      %T{type: {_, size_right}} = right
+
+      if size_left >= size_right do
+        NIF.unquote(op)(to_ref(left), to_ref(right))
+      else
+        NIF.unquote(op)(to_ref(right), to_ref(left))
+      end
+      |> from_ref(out)
+    end
+  end
+
+  unary_ops =
+    Enum.map(Nx.Shared.unary_math_funs(), &elem(&1, 0)) ++
+      [:abs, :bitwise_not, :ceil, :floor, :negate, :round, :sign]
+
+  # [:count_leading_zeros, :population_count]
+
+  for op <- unary_ops do
+    if {op, 1} in NIF.__info__(:functions) do
+      @impl true
+      def unquote(op)(out, tensor) do
+        NIF.unquote(op)(to_ref(tensor)) |> from_ref(out)
+      end
+    end
+  end
+
+  @impl true
+  def dot(
+        out,
+        %T{} = left,
+        _axes1,
+        %T{} = right,
+        _axes2
+      ) do
+    NIF.dot(to_ref(left), to_ref(right)) |> from_ref(out)
+  end
+
+  @impl true
+  def cholesky(%T{} = out, %T{} = t) do
+    NIF.cholesky(to_ref(t)) |> from_ref(out)
+  end
+
+  @impl true
+  def qr(
+        {q_holder, r_holder},
+        tensor,
+        opts
+      ),
+      do: NIF.qr(to_ref(tensor), opts[:mode] == :reduced) |> from_pair_ref({q_holder, r_holder})
+
+  @big_tensor_threshold_bytes 10_000_000
+
+  @impl true
+  def inspect(%T{type: {_, elem_size}} = tensor, inspect_opts) do
+
+    limit = if(inspect_opts.limit == :infinity, do: nil, else: inspect_opts.limit + 1)
+
+    result =
+      if on_cpu?(tensor) do
+        byte_size = nbytes(tensor)
+        byte_limit = limit && limit * div(elem_size, 8)
+
+        if min(byte_limit, byte_size) > @big_tensor_threshold_bytes do
+          "Torchx tensor is too large to inspect. Explicitly transfer the tensor by calling Nx.backend_transfer/1"
+        else
+          binary = to_blob(tensor, limit)
+          Nx.Backend.inspect(tensor, binary, inspect_opts)
+        end
+      else
+        "Tensors on the GPU cannot be inspected. Explicitly transfer the tensor by calling Nx.backend_transfer/1"
+      end
+
+    maybe_add_signature(result, tensor)
+  end
+
+  @doc """
+
+  Do not add signature during tests to match Nx doctests.
+
+  """
+  defp maybe_add_signature(result, tensor) do
+    if Mix.env() == :test do
+      result
+    else
+      Inspect.Algebra.concat(["Torchx.Backend(#{device(tensor)})", Inspect.Algebra.line(), result])
+    end
+  end
```

Comment:
Instead do this:

```suggestion
  if Application.get_env(:torchx, :add_backend_on_inspect, true) do
    defp maybe_add_signature(result, tensor) do
      Inspect.Algebra.concat(["Torchx.Backend(#{device(tensor)})", Inspect.Algebra.line(), result])
    end
  else
    defp maybe_add_signature(result, _tensor) do
      result
    end
  end
```

And then create a config/runtime.exs with this:

```elixir
import Config
config :torchx, :add_backend_on_inspect, config_env() != :test
```

Reply:
`runtime.exs` did not work, `config.exs` did.